lib/ffmpeg/filters/scale.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative '../filter'
+
+module FFMPEG
+  # rubocop:disable Style/Documentation
+  module Filters
+    # rubocop:enable Style/Documentation
+
+    class << self
+      def scale(width: nil, height: nil, force_original_aspect_ratio: nil, flags: nil)
+        Scale.new(width:, height:, force_original_aspect_ratio:, flags:)
+      end
+    end
+
+    # The Scale class uses the scale filter
+    # to resize a multimedia stream.
+    class Scale < Filter
+      class << self
+        # Returns a scale filter that fits the specified media
+        # within the specified maximum width and height,
+        # keeping the original aspect ratio.
+        #
+        # @param media [FFMPEG::Media] The media to fit.
+        # @param max_width [Numeric] The maximum width to fit.
+        # @param max_height [Numeric] The maximum height to fit.
+        # @return [FFMPEG::Filters::Scale] The scale filter.
+        def contained(media, max_width: nil, max_height: nil)
+          unless media.is_a?(FFMPEG::Media)
+            raise ArgumentError,
+                  "Unknown media format #{media.class}, expected #{FFMPEG::Media}"
+          end
+
+          if max_width && !max_width.is_a?(Numeric)
+            raise ArgumentError,
+                  "Unknown max_width format #{max_width.class}, expected #{Numeric}"
+          end
+
+          if max_height && !max_height.is_a?(Numeric)
+            raise ArgumentError,
+                  "Unknown max_height format #{max_height.class}, expected #{Numeric}"
+          end
+
+          return unless max_width || max_height
+
+          if media.rotated?
+            width = max_height || -2
```

Comment:
this is standard in ffmpeg's scale filter and I don't know if I can name the value in a constant

but basically if you specify `-2` in ffmpeg's scale filter for one of the dimensions, then ffmpeg will keep the aspect ratio and scale the image down to the other dimension while making sure the dimension where you put `-2` is even
(it also has a `-1` magic value that does the same without making sure the calculated dimension is going to be even – we're not using that here, because the libx264 encoder requires an even number for both dimensions)

IDK what I would call it, `FFMPEG::Filters::Scale::MAGIC_EVEN_DIMENSION`? I feel like that's making this code unnecessarily long

Reply:
talked about this, updated the code accordingly